crates/ruff_linter/src/rules/numpy/snapshots/ruff_linter__rules__numpy__tests__numpy2-deprecation_NPY201.py.snap

```diff
@@ -0,0 +1,865 @@
+---
+source: crates/ruff_linter/src/rules/numpy/mod.rs
+---
+NPY201.py:4:5: NPY201 [*] `np.add_docstring` will be removed in NumPy 2.0. Use `numpy.lib.add_docstring` instead.
+  |
+2 |     import numpy as np
+3 | 
+4 |     np.add_docstring
+  |     ^^^^^^^^^^^^^^^^ NPY201
+5 | 
+6 |     np.add_newdoc
+  |
+  = help: Use `numpy.lib.add_docstring` instead.
+
+ℹ Fix
+  1 |+from numpy.lib import add_docstring
+1 2 | def func():
+2 3 |     import numpy as np
+3 4 | 
+4   |-    np.add_docstring
+  5 |+    add_docstring
+5 6 | 
+6 7 |     np.add_newdoc
+7 8 | 
+
+NPY201.py:6:5: NPY201 [*] `np.add_newdoc` will be removed in NumPy 2.0. Use `numpy.lib.add_newdoc` instead.
+  |
+4 |     np.add_docstring
+5 | 
+6 |     np.add_newdoc
+  |     ^^^^^^^^^^^^^ NPY201
+7 | 
+8 |     np.add_newdoc_ufunc
+  |
+  = help: Use `numpy.lib.add_newdoc` instead.
+
+ℹ Fix
+  1 |+from numpy.lib import add_newdoc
+1 2 | def func():
+2 3 |     import numpy as np
+3 4 | 
+4 5 |     np.add_docstring
+5 6 | 
+6   |-    np.add_newdoc
+  7 |+    add_newdoc
+7 8 | 
+8 9 |     np.add_newdoc_ufunc
+9 10 | 
+
+NPY201.py:8:5: NPY201 `np.add_newdoc_ufunc` will be removed in NumPy 2.0. `add_newdoc_ufunc` is an internal function.
+   |
+ 6 |     np.add_newdoc
+ 7 | 
+ 8 |     np.add_newdoc_ufunc
+   |     ^^^^^^^^^^^^^^^^^^^ NPY201
+ 9 | 
+10 |     np.asfarray([1,2,3])
+   |
+   = help: `add_newdoc_ufunc` is an internal function.
+
+NPY201.py:10:5: NPY201 `np.asfarray` will be removed in NumPy 2.0. Use `np.asarray` with a `float` dtype instead.
+   |
+ 8 |     np.add_newdoc_ufunc
+ 9 | 
+10 |     np.asfarray([1,2,3])
+   |     ^^^^^^^^^^^ NPY201
+11 | 
+12 |     np.byte_bounds(np.array([1,2,3]))
+   |
+   = help: Use `np.asarray` with a `float` dtype instead.
+
+NPY201.py:12:5: NPY201 [*] `np.byte_bounds` will be removed in NumPy 2.0. Use `numpy.lib.array_utils.byte_bounds` instead.
+   |
+10 |     np.asfarray([1,2,3])
+11 | 
+12 |     np.byte_bounds(np.array([1,2,3]))
+   |     ^^^^^^^^^^^^^^ NPY201
+13 | 
+14 |     np.cast
+   |
+   = help: Use `numpy.lib.array_utils.byte_bounds` instead.
+
+ℹ Fix
+   1  |+from numpy.lib.array_utils import byte_bounds
```

Comment:
Sorry to be so noisy after the PR has been merged.  

I can't do this import in numpy 1.26.0. It is possible to do `from numpy.lib.utils import byte_bounds`

Reply:
Hi! This is the only backward incompatible change in the rule, mentioned in the comment: https://github.com/astral-sh/ruff/pull/7702#issuecomment-1776953773. It will be available in `array_utils` since 2.0.

Reply:
Then I'm really sorry about the noise :slightly_smiling_face:

Reply:
I should probably make this a suggested fix with a clear message.

Reply:
Fixed in https://github.com/astral-sh/ruff/pull/8474